### DISCUSSION TITLE
Toastmota: Using a plug with button for a toaster

### DISCUSSION THREAD

**Discussion by hagenbuch** [arendst/Tasmota#18769] [Q&A]
I want to use an old toaster with a Tasmota plug. Those old toasters had no time switch, so that's what I'd like to achieve with Tasmota.

Lets say I want to turn the switch on for 30 seconds when I press the button once.

However, with any successive button press, I want to add 30 seconds to the remaining time!

I guess I have to add two or three rules: 

- One that turns the switch on if a value1 is greater than the seconds since start (let's call that "time" from now on), off otherwise
- Button press rule: Add 30 seconds to value1 if value1 > time, set value1 = time + 30 seconds otherwise

What do you think? I have worked with rules before but am still a newbie. And we'll have to turn off the "toggle mode" first i guess.

Every hint is welcome

-- Andreas

**Comment by sfromis** [comment]
One approach would be to not manage it as remaining seconds, to be added to, but instead as a count of 30-seconds periods. If you use a normal PulseTime for 30 seconds, you could have a rule reacting to it going off, and in case of more cycles needed, quickly turn on again, and reduce the count. Obviously, your button push would add to that counter, and do a power on.

**Comment by hagenbuch** [comment]
Thanks! 

If I understood well, each button press would just add 1 to a multiplicator of the PulseTimer time? That may reach the same goal but I am not sure if it is possible to manipulate the pulsetimer while a time is "running"..

Hwever, this multiplicator would have to be reset just like my "value1". I guess..

My main idea is to:

- either decide from start that I want "n times 30 seconds" or else
- add more seconds later if I see the toast is not dark enough.

**Reply by sfromis** [reply]
Not needing to manipulate timer values was the main point of the suggestion. You do not need to decide from the start, and not add any seconds. Just have a counter for number of wanted 30-second cycles, count it up when getting a button push, and count it down when the pulsetime expires. And power on again immediately if the counter did not go to zero.

First push of the button should also trigger a power on, of course. The button should be decoupled from the relay, otherwise Tasmota will restart the pulsetime each push, adding some varying seconds to the total time for the cycle counter to get to zero again.

Sure, managing seconds would also be possible, just quite a bit more complicated rules than the simple setup I suggested.

**Reply by hagenbuch** [reply]
Ok I see. Could you maybe point me to an example or give me idea how those rules would have to look like and how they depend on each other?
